tests/Checkout/SalesChannel/CustomerVaultTokenRouteTest.php

```diff
@@ -0,0 +1,89 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\Test\Checkout\SalesChannel;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Test\Generator;
+use Swag\PayPal\Checkout\SalesChannel\CustomerVaultTokenRoute;
+use Swag\PayPal\DataAbstractionLayer\VaultToken\VaultTokenEntity;
+use Swag\PayPal\RestApi\V1\Api\Token;
+use Swag\PayPal\RestApi\V1\Resource\TokenResourceInterface;
+use Swag\PayPal\Test\Helper\SalesChannelContextTrait;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class CustomerVaultTokenRouteTest extends TestCase
+{
+    use SalesChannelContextTrait;
+
+    private EntityRepository&MockObject $repository;
+
+    private TokenResourceInterface&MockObject $tokenResource;
+
+    private CustomerVaultTokenRoute $route;
+
+    protected function setUp(): void
+    {
+        $this->repository = $this->createMock(EntityRepository::class);
+        $this->tokenResource = $this->createMock(TokenResourceInterface::class);
+
+        $this->route = new CustomerVaultTokenRoute($this->repository, $this->tokenResource);
+    }
+
+    public function testGetVaultTokenWithoutCustomer(): void
+    {
+        $salesChannelContext = Generator::generateSalesChannelContext();
+        $salesChannelContext->assign(['customer' => null]);
+
+        $response = $this->route->getVaultToken($salesChannelContext);
+
+        static::assertNull(
+            \json_decode((string) $response->getContent(), true, 512, \JSON_THROW_ON_ERROR)['token']
+        );
```

Comment:
for more readability if this test ever fails:
```suggestion
        static::assertSame(
            '{"token":null}'
            (string) $response->getContent()
        );
```